### PR TITLE
Add Data Studio 5G overlay file system limit

### PR DESCRIPTION
**Pull request by jason-seqera**
First pass, needs revisions. Request originated on Slack: https://seqera.slack.com/archives/C053KDKLEQJ/p1714947936506089

### CODE REVIEW

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx [hunk 1]
```diff
@@ -167,3 +167,9 @@ If you have access to the AWS Console for your organization, check the jobs asso
 ### What happens if the memory allocation of the data studio is exceeded?
 
 The running container in the AWS Batch compute environment inherits the memory limits specified by the data studio configuration when adding or starting the session. The kernel then handles the memory as if running natively on Linux. Linux can overcommit memory, leading to possible out-of-memory errors in a container environment. The kernel has protections in place to prevent this, but it can happen, and in this case, the process is killed. This can manifest as a performance lag, killed subprocesses, or at worst, a killed data studio session.
+
+### What happens when the allocated storage for the data studio is exceeded?
+
+A data studio is allocated a 5G overlay filesystem partition for installing software packages and other local configuration. If you exceed the available storage space, you cannot reliably make local changes to the data studio until you free up disk space.
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:173]:
```suggestion
Currently, a data studio is allocated a 5GB overlay filesystem partition by default for installing software packages and other local configuration. This limit is irrespective of the boot disk size of the container. If you exceed the available storage space, you cannot reliably make local changes to the data studio container until you free up disk space.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx [hunk 2]
```diff
@@ -167,3 +167,9 @@ If you have access to the AWS Console for your organization, check the jobs asso
...
+
+Each time you stop and start a data studio, an additional 5G allocation is provided. As a workaround, you can stop your data studio session, and then start the data studio again to have access to 5G of disk space.
```
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:175]:
```suggestion
Each time you stop and start a data studio, an additional 5GB allocation is provided. If you are experiencing a disk size limitation, as a workaround you can stop and restart the data studio.
```

**Review by robnewman — CHANGES_REQUESTED** [review]

**Comment by jason-seqera** [comment]
@robnewman, we've decided to follow Google Developer Style for this, so I've omitted _currently_ for consistency with that. When do we want to merge this? Thanks!

**Review by robnewman — APPROVED** [review]

**Comment by jason-seqera** [comment]
I'm going to merge this as Rob approved this on Slack DM. Thanks!

**Pull request merged**